src/reducers/narrative.ts

```diff
@@ -1,19 +1,47 @@
 import * as types from "../actions/types";
+import { AnyAction } from 'redux';
 
-const narrative = (state = {
+export interface NarrativeState {
+  loaded: boolean
+  /**
+   * array of paragraphs (aka blocks)
+   */
+  blocks: { __html: string }[] | null
+
+  /**
+   * which block is currently "in view"
+   */
+  blockIdx?: number
+
+  /**
+   * the pathname of the _narrative_
+   */
+  pathname?: string
+
+  display: boolean
+  title?: string
+}
+
+const defaultState: NarrativeState = {
   loaded: false,
-  blocks: null, /* array of paragraphs (aka blocks) */
-  blockIdx: undefined, /* which block is currently "in view" */
-  pathname: undefined,  /* the pathname of the _narrative_ */
+  blocks: null,
+  blockIdx: undefined,
+  pathname: undefined,
   display: false,
   title: undefined
-}, action) => {
+};
+
+const narrative = (
+  state: NarrativeState = defaultState,
+  action: AnyAction,
```

Comment:
can you provide a default value to a positional param if there's a subsequent positional param without a default value? or do you need to switch the order here, or convert this to being passed an object?

Reply:
Good point – ESLint and TypeScript clearly had no complaints about this so I searched to confirm – it is allowed in JavaScript. From [MDN docs on default parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters):

> Parameters are still set left-to-right, overwriting default parameters even if there are later parameters without defaults.
>
> ```js
> function f(x = 1, y) {
>   return [x, y];
> }
> 
> f(); // [1, undefined]
> f(2); // [2, undefined]
> ```

Reply:
I don't think it's a good pattern, but it is the pattern used for other files in `src/reducers/` so I will keep it as-is.

Reply:
Just a quick comment on this (and similar ones such as `const foo = function foo() {...}`) - if you see stuff that is clearly poorly implemented it's probably because it was just a poor implementation. Please add a FIXME comment in the code or just fix it. I'd prefer inconsistent code over consistently bad code.